greeter/README.md

```diff
@@ -1,21 +1,727 @@
+   ```json
+   {
+     "timestamp": "2019-10-01T18:47:37.096Z",
+     "language": "fr",
+     "greeting": "Diana, Bonjour"
+   }
+   ```
+
+### Step 4: Configure the Greeter
+
+Besides enabling you to bind various types of classes to a binding key on a context, LB4 also allows you to configure bound items easily. Such configurations can be resolved and injected in the same way as other dependencies.
+
+The default setting in `ChineseGreeter` is to say your `name` first, followed by `hello`. For example `"LoopBack, hello"`. Let's `configure` the `ChineseGreeter` to say `hello` followed by your `name`.
+
+Add this line in your `application.ts` file
+
+```ts
+this.component(GreetingComponent);
+this.add(createBindingFromClass(FrenchGreeter));
+
+// Add the following line
+this.configure('greeters.ChineseGreeter').to({nameFirst: false});
+```
+
+### Step 5: Try it Out!
+
+Restart your application, try endpoint `GET /greeting/{name}`, with `language` set to `zh`.
+
+You will see message `'你好, name'` printed instead of `'name, 你好'`.
+
+Related documentation:
+[Configuration by convention](https://loopback.io/doc/en/lb4/Context.html#configuration-by-convention)
+
+## Part 3: Enable Caching
+
+We have showed in the previous steps that how to decouple your application and design it as a microservice. To enhance the idea of building a microservice, we will add more services to the greeting app.
+
+To reduce the cost of calculation, we'd like to use a caching service for the greeting application. We will show how to bind a caching service, interceptor and observer to the app.
+
+The caching service will cache the greeting messages for a certain amount of time. And cached messages will be swept out when they are expired.
+
+Again, we are using `bind` and `injection` to bind our app with an Interceptor and Observer:
+
+- Interceptors: are reusable functions that provide [aspect-oriented](https://en.wikipedia.org/wiki/Aspect-oriented_programming) logic around method invocations.
+- Observer: is a LoopBack application that has its own life cycles at runtime. An [`Observer allows`](https://loopback.io/doc/en/lb4/Life-cycle.html#the-lifecycleobserver-interface) artifacts to participate in the life cycles. The transition of states is triggered by two functions: `start()` and `stop()`.
+
+### Step 1: Create a caching service
+
+1.  Copy the contents of `caching.service.ts` to a new file in your workspace named `src/caching.service.ts`.
+
+2.  Create the binding key for caching service.
+
+As we mentioned above, a binding connects its value to a unique key as the address to access the entry in a context. For example,
+
+```ts
+ctx.bind('hello').to('world'); // BindingKey='hello', BindingValue='world'`
+```
+
+Let's create the binding key for the caching service so that you can inject it in an observer or an interceptor later.
+
+        In `src` folder, create a file called `keys.ts` with the following content:
```

Comment:
not sure if it's the extra indent, the "in src folder..." is shown as code snippet in the view.  feel free to fix it in a subsequent PR.